core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -251,10 +257,43 @@ class JDBCSourceAsColumnarStore(override val connProperties: ConnectionPropertie
   override def getConnection(id: String, onExecutor: Boolean): Connection = {
     val connProps = if (onExecutor) connProperties.executorConnProps
     else connProperties.connProps
-    ConnectionPool.getPoolConnection(id, connProperties.dialect,
+    getPoolConnection(id, connProperties.dialect,
       connProperties.poolProps, connProps, connProperties.hikariCP)
   }
 
+  private lazy val bootProperties = Option(Misc.getMemStoreBootingNoThrow).map(_.getBootProperties)
+
+  def getPoolConnection(id: String, dialect: JdbcDialect, poolProps: Map[String, String],
+      connProps: Properties, hikariCP: Boolean): Connection = {
+    // Handle security at remote VM in cases like insert
+    bootProperties match {
+      case Some(p) if p.containsKey(Attribute.USERNAME_ATTR) &&
+          p.containsKey(Attribute.PASSWORD_ATTR) =>
+        def secureProps(props: Properties): Properties = {
+          if (Option(props.getProperty(Attribute.USERNAME_ATTR)).isEmpty &&
+              Option(props.getProperty(Attribute.PASSWORD_ATTR)).isEmpty) {
+            props.setProperty(Attribute.USERNAME_ATTR, p.get(Attribute.USERNAME_ATTR).toString)
+            props.setProperty(Attribute.PASSWORD_ATTR, p.get(Attribute.PASSWORD_ATTR).toString)
+          }
+          props
+        }
+
+        // Hikari only take 'username'. So does Tomcat
+        def securePoolProps(props: Map[String, String]): Map[String, String] = {
+          if (props.get(Attribute.USERNAME_ALT_ATTR.toLowerCase).isEmpty &&
+              props.get(Attribute.PASSWORD_ATTR).isEmpty) {
+            props + (Attribute.USERNAME_ALT_ATTR.toLowerCase ->
+                p.get(Attribute.USERNAME_ATTR).toString) +
+                (Attribute.PASSWORD_ATTR -> p.get(Attribute.PASSWORD_ATTR).toString)
+          } else props
+        }
```

Comment:
Hi @sumwale , this was needed for allowing inserts on column table. Inserts convert in put/putAll and and they do not carry credentials to remote node. I evaluated the efforts after discussions with @kneeraj and this looked safe (since connection is internal) and quick. Alternative was to carry credentials in messages like Put/PutAll and Batchupdate than can be done after 1.0 if you feel its needed. Kindly suggest.